### PR TITLE
HIP cleanup for version 3.5

### PR DESCRIPTION
**Pull request by masterleinad**
The changes in this pull request are incompatible with HIP 3.3.

### CODE REVIEW

**Comment by dalg24** [comment]
Please rebase on develop now that we merged #3094

**Comment by masterleinad** [comment]
> Please rebase on develop now that we merged #3094

Done.

**Review by crtrott — APPROVED** [review]

**Comment by masterleinad** [comment]
Retest this please.

File: core/unit_test/hip/TestHIP_TeamReductionScan.cpp [hunk 1]
```diff
@@ -47,17 +47,14 @@
...
-//  TestScanTeam<TEST_EXECSPACE, Kokkos::Schedule<Kokkos::Dynamic> >(10000);
-//}
-//#endif
+TEST(TEST_CATEGORY, team_scan) {
+  TestScanTeam<TEST_EXECSPACE, Kokkos::Schedule<Kokkos::Static> >(0);
+  TestScanTeam<TEST_EXECSPACE, Kokkos::Schedule<Kokkos::Dynamic> >(0);
+  TestScanTeam<TEST_EXECSPACE, Kokkos::Schedule<Kokkos::Static> >(10);
+  TestScanTeam<TEST_EXECSPACE, Kokkos::Schedule<Kokkos::Dynamic> >(10);
+  //  TestScanTeam<TEST_EXECSPACE, Kokkos::Schedule<Kokkos::Static> >(10000);
+  //  TestScanTeam<TEST_EXECSPACE, Kokkos::Schedule<Kokkos::Dynamic> >(10000);
```
Comment (dalg24) [inline core/unit_test/hip/TestHIP_TeamReductionScan.cpp:56]: Why are the last two commented out?

Comment (masterleinad) [reply core/unit_test/hip/TestHIP_TeamReductionScan.cpp:56]: Oh, they still don't work. Let me reintroduce `FIXME_HIP` there.

**Comment by masterleinad** [comment]
Retest this please.

**Pull request merged**